### ISSUE TITLE
Tagged libraries fail?

### ISSUE THREAD

**Issue by jbae11** [sskutnik/cyborg#23]
@sskutnik 
Error Message with libraries?
arp2obl.py script was (i think) successful ([here](
https://gist.github.com/jbae11/2d0af3aaaa87982406270326c1feea05))
```
dkadkf@arfc-100 ~/github/cyborg/inputs/MOX-BWR (master) $ cyclus MOX_src_1_Reactor_1_year.xml 
              :                                                               
          .CL:CC CC             _Q     _Q  _Q_Q    _Q    _Q              _Q   
        CC;CCCCCCCC:C;         /_\)   /_\)/_/\\)  /_\)  /_\)            /_\)  
        CCCCCCCCCCCCCl       __O|/O___O|/O_OO|/O__O|/O__O|/O____________O|/O__
     CCCCCCf     iCCCLCC     /////////////////////////////////////////////////
     iCCCt  ;;;;;.  CCCC                                                      
    CCCC  ;;;;;;;;;. CClL.                          c                         
   CCCC ,;;       ;;: CCCC  ;                   : CCCCi                       
    CCC ;;         ;;  CC   ;;:                CCC`   `C;                     
  lCCC ;;              CCCC  ;;;:             :CC .;;. C;   ;    :   ;  :;;   
  CCCC ;.              CCCC    ;;;,           CC ;    ; Ci  ;    :   ;  :  ;  
   iCC :;               CC       ;;;,        ;C ;       CC  ;    :   ; .      
  CCCi ;;               CCC        ;;;.      .C ;       tf  ;    :   ;  ;.    
  CCC  ;;               CCC          ;;;;;;; fC :       lC  ;    :   ;    ;:  
   iCf ;;               CC         :;;:      tC ;       CC  ;    :   ;     ;  
  fCCC :;              LCCf      ;;;:         LC :.  ,: C   ;    ;   ; ;   ;  
  CCCC  ;;             CCCC    ;;;:           CCi `;;` CC.  ;;;; :;.;.  ; ,;  
    CCl ;;             CC    ;;;;              CCC    CCL                     
   tCCC  ;;        ;; CCCL  ;;;                  tCCCCC.                      
    CCCC  ;;     :;; CCCCf  ;                     ,L                          
     lCCC   ;;;;;;  CCCL                                                      
     CCCCCC  :;;  fCCCCC                                                      
      . CCCC     CCCC .                                                       
       .CCCCCCCCCCCCCi                                                        
          iCCCCCLCf                                                           
           .  C. ,                                                            
              :                                                               
Experimental Warning: MatlSellPolicy is experimental and its API may be subject to change
Experimental Warning: The CyBORG Reactor is highly experimental.
Experimental Warning: MatlSellPolicy is experimental and its API may be subject to change
Experimental Warning: The CyBORG Reactor is highly experimental.
Experimental Warning: MatlSellPolicy is experimental and its API may be subject to change
Experimental Warning: The CyBORG Reactor is highly experimental.
Experimental Warning: MatlSellPolicy is experimental and its API may be subject to change
Experimental Warning: The CyBORG Reactor is highly experimental.
Experimental Warning: MatlSellPolicy is experimental and its API may be subject to change
Experimental Warning: The CyBORG Reactor is highly experimental.
Interp tag name: Plutonium Content, value: 4.50005.
Interp tag name: Plutonium-239 Content, value: 61.9993.
Key Warning: Recipe lookup failed: Invalid recipe name {
   "Assembly Type" : "mox_ge10x10-8",
   "Moderator Density" : 0.550,
   "Plutonium Content" : 4.500050000000001,
   "Plutonium-239 Content" : 61.99931111876536,
   "Powers (W)" : "3.33333e+07,0,3.33333e+07,0,3.33333e+07",
   "Times (y)" : "0,1,1.08333,2.08333,2.16667,3.16667"
}

Cyborg::reactor::interpolate(937) : No libraries found that match specified ID tags!
```

Do you have any ideas what might be wrong?

**Comment by sskutnik** [comment]
Hi @jbae11 - I presume based on the recipe you're looking at the MOX_src_1_Reactor_1_year.xml input in  /inputs/MOX-BWR, correct?

For whatever reason, it seems like CyBORG is not finding your tagged libraries. This could be for one of a couple reasons. For starters, could you tell me what the path in src/orglib_default_location points to (and if this is consistent with the location of your tagged Origen reactor libraries?)

**Comment by jbae11** [comment]
```#define ORIGEN_LIBS_DEFAULT "/home/dkadkf/SCALE/SCALE-6.2-serial-6.2.2-Source/data/arplibs"```

and yes, this is where my tagged Origen reactor libraries, if I am not mistaken. 

Thank you

**Comment by nicholas-sly** [comment]
It has been a while since I've run the script, but I thought it would print
out the tags every time it successfully applied changes to each library.
It might be worth using `obiwan tag [library-name]` to verify that the tags
have been properly applied.  You can use a wildcard to point that call at
all of the mox libraries you're using.  This should tell you what tags
those libraries have, what name and values it expects, and if there are any
errors that have popped up but been otherwise suppressed.

Let us know what that reveals.

Thanks.

**Comment by jbae11** [comment]
Yeah, now it shows errors [here](https://gist.github.com/jbae11/7a56435eac2b55fa461861c3ade0ab67).

that is odd, though. Do you have an idea why it says that it is not an Origen::Library?

**Comment by sskutnik** [comment]
So judging from the script output, it looks like you're running arp2obl.py from the ```${CYBORG}/scripts``` directory, and it's trying to tag libraries in that directory and subsequently failing.

Pulling from the first couple of lines of your output:

```
~/github/cyborg/scripts (master) $ python2 arp2obl.py --obiwan ~/SCALE/SCALE-6.2-serial-6.2.2-Source/build/gcc/INSTALL/bin/obiwan --arpdata ~/SCALE/SCALE-6.2-serial-6.2.2-Source/data/arpdata.txt
ORIGEN Library directory used:  .
```
i.e., without giving arp2obl.py a target directory to work off of, it doesn't find any libraries. (Note that it does find ```arpdata.txt```, so it knows what to expect; it just doesn't find the actual Origen reactor data libraries.)

Make sure you give it the ```--libdir``` flag when you use it, i.e.:
```
/path/to/origen/library/directory/
```

**Comment by jbae11** [comment]
Interesting. I looked at the script and I don't think `--libdir` changes anything. Using the flag does change the `libdir` variable, but it isn't used anywhere, to change to reference directory.

However, I was able to successfully tag the libraries by running the script in the `arplib` directory, like such:
```
dkadkf@arfc-100 ~/SCALE/SCALE-6.2-serial-6.2.2-Source/data/arplibs $ python2 ~/github/cyborg/scripts/arp2obl.py --arpdata ~/SCALE/SCALE-6.2-serial-6.2.2-Source/data/arpdata.txt --obiwan ~/SCALE/SCALE-6.2-serial-6.2.2-Source/build/gcc/INSTALL/bin/obiwan
```

I ran an example from the input file to test if this works:
```
dkadkf@arfc-100 ~/github/cyborg/inputs/MOX-BWR (master) $ cyclus MOX_src_1_Reactor_1_year.xml 
              :                                                               
          .CL:CC CC             _Q     _Q  _Q_Q    _Q    _Q              _Q   
        CC;CCCCCCCC:C;         /_\)   /_\)/_/\\)  /_\)  /_\)            /_\)  
        CCCCCCCCCCCCCl       __O|/O___O|/O_OO|/O__O|/O__O|/O____________O|/O__
     CCCCCCf     iCCCLCC     /////////////////////////////////////////////////
     iCCCt  ;;;;;.  CCCC                                                      
    CCCC  ;;;;;;;;;. CClL.                          c                         
   CCCC ,;;       ;;: CCCC  ;                   : CCCCi                       
    CCC ;;         ;;  CC   ;;:                CCC`   `C;                     
  lCCC ;;              CCCC  ;;;:             :CC .;;. C;   ;    :   ;  :;;   
  CCCC ;.              CCCC    ;;;,           CC ;    ; Ci  ;    :   ;  :  ;  
   iCC :;               CC       ;;;,        ;C ;       CC  ;    :   ; .      
  CCCi ;;               CCC        ;;;.      .C ;       tf  ;    :   ;  ;.    
  CCC  ;;               CCC          ;;;;;;; fC :       lC  ;    :   ;    ;:  
   iCf ;;               CC         :;;:      tC ;       CC  ;    :   ;     ;  
  fCCC :;              LCCf      ;;;:         LC :.  ,: C   ;    ;   ; ;   ;  
  CCCC  ;;             CCCC    ;;;:           CCi `;;` CC.  ;;;; :;.;.  ; ,;  
    CCl ;;             CC    ;;;;              CCC    CCL                     
   tCCC  ;;        ;; CCCL  ;;;                  tCCCCC.                      
    CCCC  ;;     :;; CCCCf  ;                     ,L                          
     lCCC   ;;;;;;  CCCL                                                      
     CCCCCC  :;;  fCCCCC                                                      
      . CCCC     CCCC .                                                       
       .CCCCCCCCCCCCCi                                                        
          iCCCCCLCf                                                           
           .  C. ,                                                            
              :                                                               
Experimental Warning: MatlSellPolicy is experimental and its API may be subject to change
Experimental Warning: The CyBORG Reactor is highly experimental.
Experimental Warning: MatlSellPolicy is experimental and its API may be subject to change
Experimental Warning: The CyBORG Reactor is highly experimental.
Experimental Warning: MatlSellPolicy is experimental and its API may be subject to change
Experimental Warning: The CyBORG Reactor is highly experimental.
Experimental Warning: MatlSellPolicy is experimental and its API may be subject to change
Experimental Warning: The CyBORG Reactor is highly experimental.
Experimental Warning: MatlSellPolicy is experimental and its API may be subject to change
Experimental Warning: The CyBORG Reactor is highly experimental.
Interp tag name: Plutonium Content, value: 4.50005.
Interp tag name: Plutonium-239 Content, value: 61.9993.
Key Warning: Recipe lookup failed: Invalid recipe name {
   "Assembly Type" : "mox_ge10x10-8",
   "Moderator Density" : 0.550,
   "Plutonium Content" : 4.500050000000001,
   "Plutonium-239 Content" : 61.99931111876536,
   "Powers (W)" : "3.33333e+07,0,3.33333e+07,0,3.33333e+07",
   "Times (y)" : "0,1,1.08333,2.08333,2.16667,3.16667"
}

Interp tag name: Moderator Density, value: 0.55.
Interp tag name: Plutonium Content, value: 4.50005.
Interp tag name: Plutonium-239 Content, value: 61.9993.

Status: Cyclus run successful!
Output location: cyclus.sqlite
Simulation ID: 617d309b-b6bb-4782-b156-eaf88a3e1dce
```
though it says recipe lookup fail, the output fuel recipe ( I had to add a sink in the input file to see it) seemed okay. Do you think this is a problem?

**Comment by sskutnik** [comment]
So the warning message you got:
```
Key Warning: Recipe lookup failed: Invalid recipe name {
   "Assembly Type" : "mox_ge10x10-8",
   "Moderator Density" : 0.550,
   "Plutonium Content" : 4.500050000000001,
   "Plutonium-239 Content" : 61.99931111876536,
   "Powers (W)" : "3.33333e+07,0,3.33333e+07,0,3.33333e+07",
   "Times (y)" : "0,1,1.08333,2.08333,2.16667,3.16667"
}
```
is essentially just a Cyclus warning that says this unique combination defining this recipe has not yet been defined in the simulation runtime. (What this means for you is that CyBORG needs to run a depletion case to generate this recipe the first time it encounters it). Further instances of this irradiation history will now show up in Cyclus' recipe cache, meaning that the depletion calculation is not re-run.
In other words - it's not a problem, simply a warning that Cyclus doesn't (yet) know about this recipe.

**Comment by jbae11** [comment]
Cool! thanks! I'll close this issue now.
